### ISSUE TITLE
Problem with newsletter_email_to_user

### ISSUE THREAD

**Issue by dasistwas** [dasistwas/moodle-mod_newsletter#35]
The way Moodle sends emails has changed. Especially the EMAIL_VIA_ALWAYS feature is not respected in the current version of newsletter_email_to_user.

**How to solve:**

1) Copy Moodle core email_to_user from lib/moodlelib.php in newsletter_email_to_user and then make the changes necessary for the newsletter module. So you have the new way of sending mails from Moodle and the newsletter specific code changes.

**Comment by dasistwas** [comment]
// Make up an email address for handling bounces.
        if (! empty ( $CFG->handlebounces )) {
                $modargs = 'B' . base64_encode ( pack ( 'V', $user->id ) ) . substr ( md5 ( $user->email ), 0, 16 );
                $mail->Sender = generate_email_processing_address ( 0, $modargs );
        } else {
-               $mail->Sender = $CFG->noreplyaddress;
+               if($bounceemail != ''){
+                       $mail->Sender = $bounceemail;
+               } else {
+                       $mail->Sender = $CFG->noreplyaddress;
+               }
        }

**Comment by dasistwas** [comment]
`        // Make up an email address for handling bounces.
        if (! empty ( $CFG->handlebounces )) {
                $modargs = 'B' . base64_encode ( pack ( 'V', $user->id ) ) . substr ( md5 ( $user->email ), 0, 16 );
                $mail->Sender = generate_email_processing_address ( 0, $modargs );
        } else {
-               $mail->Sender = $CFG->noreplyaddress;
+               if($bounceemail != ''){
+                       $mail->Sender = $bounceemail;
+               } else {
+                       $mail->Sender = $CFG->noreplyaddress;
+               }
        }
`

**Comment by dasistwas** [comment]
```
        // Make up an email address for handling bounces.
        if (! empty ( $CFG->handlebounces )) {
                $modargs = 'B' . base64_encode ( pack ( 'V', $user->id ) ) . substr ( md5 ( $user->email ), 0, 16 );
                $mail->Sender = generate_email_processing_address ( 0, $modargs );
        } else {
-               $mail->Sender = $CFG->noreplyaddress;
+               if($bounceemail != ''){
+                       $mail->Sender = $bounceemail;
+               } else {
+                       $mail->Sender = $CFG->noreplyaddress;
+               }
        }

```

**Comment by dasistwas** [comment]
```
 
-    $supportuser = generate_email_supportuser();
+    $supportuser = core_user::get_support_user() ;
 

```

**Comment by dasistwas** [comment]
Not sure, if we still need newsletter_email_to_user or if we just pass another $from param to email_to_user

**Comment by dasistwas** [comment]
Ok, we should use email_to_user and as $from the support user.

**Comment by dasistwas** [comment]
There was also a feature added to use multiple attachments. So I updated with new email_to_user and implemented multiple attachments.

**Comment by dasistwas** [comment]
OK fixed in cb1a84d90f401611cdfbb458ee1e3542fcd225da
